jordi-villa-test/todo-react
# part-1 :: Initial Todo App (react+redux+react-redux)

**Description**

This pull request includes the first iteration of todo-react. 
It includes a development of an initial version of the app using the following stack:

- React
- Redux
- React-Redux
- Styled Components
- React Testing Library

**This PR implements**

- Eslint+Prettier+Husky implementation to ensure a correct code formatting from the initial stages of the project
- Installation of basic dependencies (react, redux, jest, styled-components...)
- Implementation of reusable components
  - Todo component
  - Button component with basic hover styling
  - TextField component
  - Svg component for icons
  - Header component
- Implementation of todo listing
  - redux reducer and selector (selectTodos)
- Implementation of basic CRUD for the todos via redux actions
  - addTodo
  - removeTodo
  - editTodo
  - completeTodo
- Implemented responsive styling 
- Implemented persisting of redux state in localStorage
- Add basic `README.md` file

**How to test**

1. Run all the unit tests in the project with `yarn test` or `npm run test`
2. Run the project following readme instructions
3. Try to add multiple todos and then editting, removing or clicking on them to complete them
4. Refresh the page and check that all todos remain in their previous state
